## Tests statistiques sur la probabilité de naitre homme ou femme.
* Echantillon1: 399 284 hommes contre 381 883
* Test d'équiprobabilité

##  Q1

In [18]:
n_homme = 399284
n_femme = 381883

In [19]:
n = n_homme + n_femme
p = n_homme / n
print("proportion d'hommes:",round(p,2))

proportion d'hommes: 0.51


On test ici une proportion : Loi de Bernouilli avec p = 0.51

In [20]:
from statsmodels.stats.proportion import binom_test

p_0 = 0.5

"""
input = nb success, nb failed, p_0 et type de test (bilatere, unilatere etc.)
"""
p_value = binom_test(n_homme,n, prop=p_0, alternative='larger')

print('p-valeur:',p_value)
print('rounded p-valeur:',round(p_value,2))

p-valeur: 1.3741661348079848e-86
rounded p-valeur: 0.0


## Q2
P_valeur de 0%, inférieure à 5%. On rejette donc l'hypothèse nulle au niveau de test 5%.

## Q3
* échantillon de taille n = 100
* échantillon de taille n = 1000
* échantillon de taille n = 10000

In [21]:
n=10
n_homme = p*n
p_value = binom_test(n_homme,n, prop=p_0, alternative='larger')

print('n=10, p-valeur:',p_value)

n=10, p-valeur: 0.6230468749999999


In [22]:
n=100
n_homme = p*n
p_value = binom_test(n_homme,n, prop=p_0, alternative='larger')

print('n=100 p-valeur:',p_value)

n=100 p-valeur: 0.4602053813064103


In [23]:
n=1000
n_homme = p*n
p_value = binom_test(n_homme,n, prop=p_0, alternative='larger')

print('n=1000, p-valeur:',p_value)

n=1000, p-valeur: 0.2533300213124749


In [25]:
n=399284+381883
n_homme = p*n
p_value = binom_test(n_homme,n, prop=p_0, alternative='larger')

print('n=',n,', p-valeur:',p_value)

n= 781167 , p-valeur: 1.436938781696969e-86


## Q4
* n=10 : p_valeur = 62%, supérieure à 5%. On ne rejette donc pas l'hypothèse nulle au niveau de test 5%.
* n=100 : p_valeur = 46%, supérieure à 5%. On ne rejette donc pas l'hypothèse nulle au niveau de test 5%.
* n=1000 : p_valeur = 25%, supérieure à 5%. On ne rejette donc pas l'hypothèse nulle au niveau de test 5%.